src/webots/nodes/WbCamera.cpp

```diff
@@ -533,6 +533,15 @@ void WbCamera::writeConfigure(QDataStream &stream) {
 
 void WbCamera::writeAnswer(QDataStream &stream) {
   WbAbstractCamera::writeAnswer(stream);
+
+  if (!mImageWritten) {
+    if (isEnabled()) {
+      copyImageToSharedMemory(mWrenCamera, image());
+      mImageChanged = false;
+    }
+    mImageReady = true;
+  }
+
```

Comment:
We should probably do this in Lidar and RangeFinder as well.
So, it could be better to put this code in `WbAbstractCamera.cpp` instead, isn't it?